xbmc/platform/darwin/tvos/XBMCController.mm

```diff
@@ -193,55 +195,46 @@ - (void)disableBackGroundTask:(UIBackgroundTaskIdentifier)bgTaskID;
 
 - (void)becomeInactive
 {
-  // if we were interrupted, already paused here
-  // else if user background us or lock screen, only pause video here, audio keep playing.
-  if (g_application.GetAppPlayer().IsPlayingVideo() && !g_application.GetAppPlayer().IsPaused())
-  {
-    m_isPlayingBeforeInactive = YES;
-    m_lastUsedPlayer = g_application.GetAppPlayer().GetCurrentPlayer();
-    m_playingFileItemBeforeBackground =
-        std::make_unique<CFileItem>(g_application.CurrentFileItem());
-    CApplicationMessenger::GetInstance().SendMsg(TMSG_MEDIA_PAUSE_IF_PLAYING);
-    g_application.CurrentFileItem().m_lStartOffset = g_application.GetAppPlayer().GetTime() - 2.50;
-  }
+  CLog::Log(LOGDEBUG, "%s",  __PRETTY_FUNCTION__);
+  CApplicationMessenger::GetInstance().SendMsg(TMSG_MEDIA_PAUSE_IF_PLAYING);
 }
 
 - (void)enterBackground
 {
+  CLog::Log(LOGDEBUG, "%s",  __PRETTY_FUNCTION__);
+
+  // We always come from becomeInactive() so a media might be in pause state
+  // Unpause the media so that PowerManager.OnSleep() can save StorePlayerState()
+  CApplicationMessenger::GetInstance().SendMsg(TMSG_MEDIA_UNPAUSE);
```

Comment:
I have my concerns about this. We have a limited time when backgrounded to execute, and using sendMsg or PostMsg doesnt give us the ability to have that thread in the background task. ie the background task (and therefore the app) may end before the msg is processed, so its an uncertain state.
As its only the code in the background task that can execute past the thread time limit (5s), i found that often the state just wasnt what you would expect when you used the msg methods.

I think it might be best to explicitly check if paused and then unpause inside the background task a few lines down.